llpc/util/llpcError.cpp

```diff
@@ -117,6 +108,26 @@ std::error_code resultToErrorCode(Result result) {
   return {static_cast<int>(result), GlobalCategoryId};
 }
 
+// =====================================================================================================================
+// Prints the error message in `err` to LLPC_ERRS and consumes the error.
+//
+// @param err : The error to handle. This must not be an `ErrorSuccess`.
+// @returns: The underlying `Result` when `err` is a `ResultError`, `Result::ErrorUnknown` otherwise.
```

Comment:
Can it ever happen that ErrorUnknown is returned? From what I understand from the LLVM docs, using handleAllErrors causes the program to terminate if the error cannot be matched.

From the docs: "[...] `handleAllErrors` function can be used instead. This is identical to `handleErrors` except that it will terminate the program if an unhandled error is passed in"

Reply:
This should happen when the Error is of some unrelated type, e.g., `StringError` or `FileError`. Because all errors must derive from `ErrorInfoBase`, the last case should handle that.

But that's a good point, I think this deserves a unit test. I'll add one.